connect/runtime/src/main/java/org/apache/kafka/connect/runtime/WorkerConfig.java

```diff
@@ -375,4 +386,31 @@ public WorkerConfig(ConfigDef definition, Map<String, String> props) {
         logInternalConverterDeprecationWarnings(props);
     }
 
+    private static class AdminListenersValidator implements ConfigDef.Validator {
+        @Override
+        public void ensureValid(String name, Object value) {
+            if (value == null) {
+                return;
+            }
+
+            if (!(value instanceof List)) {
+                throw new ConfigException("Invalid value type (list expected).");
+            }
+
+            List items = (List) value;
+            if (items.isEmpty()) {
+                return;
+            }
+
+            for (Object item: items) {
+                if (!(item instanceof String)) {
+                    throw new ConfigException("Invalid type for admin listener (expected String).");
+                }
+                if (((String) item).length() == 0) {
```

Comment:
What about the following to also prevent whitespace-only strings:
```suggestion
                if (((String) item).trim().isEmpty()) {
```